plugins/search-common/src/types.ts

```diff
@@ -66,7 +85,13 @@ export interface IndexableDocument {
    * is clicked).
    */
   location: string;
+}
 
+/**
+ * Properties related to indexing of documents.
+ * @beta
+ */
+export type IndexableDocument = SearchDocument & {
```

Comment:
What do you think of moving the `Indexable*` types out of `@backstage/plugin-search-common` and into `@backstage/plugin-search-backend-node`?  Make it super clear that those should only be used in backends.

Reply:
Yeah, I'd say let's rip the bandaid off.  Otherwise, we'll lose the context of this split and it'll be non-obvious how to reason about when to use which types.

Reply:
Makes sense. Unfortunately though, I just started this and I think like it would force a breaking change in catalog-common too, since the `CatalogEntityDocument` type would need to move to catalog-backend to avoid catalog-common (which is isomorphic) depending on search-backend (which is backend only).

Reply:
Mmmmmmm. That makes sense.

Maybe just a comment on the indexable types to indicate "this should only ever be used in backends" and then let's ship it!

Reply:
Good idea! Done in d48122a439.